src/elementPlugins/shortText/src/ApplicationView.tsx

```diff
@@ -7,52 +7,25 @@ const ApplicationView: React.FC<ApplicationViewProps> = ({
   onUpdate,
   initialValue,
   isEditable,
-  isRequired,
-  allResponses,
-  evaluator,
+  currentResponse,
+  validationState,
+  onSave,
 }) => {
-  const [validationMessageDisplay, setValidationMessageDisplay] = useState('')
-  const [value, setValue] = useState(initialValue?.text)
-  const [isValid, setIsValid] = useState<boolean>()
-  const [responses, setResponses] = useState({
-    thisResponse: undefined,
-    ...allResponses,
-  })
-  const {
-    validation: validationExpression,
-    validationMessage,
-    placeholder,
-    maskedInput,
-    label,
-  } = templateElement?.parameters
+  const [value, setValue] = useState<string>(initialValue?.text)
+  const { placeholder, maskedInput, label } = templateElement.parameters
 
   useEffect(() => {
-    // Do validation, setIsValid
-    if (validationExpression && responses.thisResponse !== undefined) {
-      evaluator(validationExpression, { objects: [responses], APIfetch: fetch })
-        .then((result: boolean) => {
-          setIsValid(result)
-        })
-        .catch((err: any) => console.log(err))
-    } else setIsValid(true)
-  }, [responses])
-
-  useEffect(() => {
-    setResponses({ thisResponse: value, ...allResponses })
-  }, [value])
-
-  useEffect(() => {
-    if (isValid) setValidationMessageDisplay('')
-    else setValidationMessageDisplay(validationMessage)
-  }, [isValid])
+    onUpdate(value)
```

Comment:
Can and should we move this to wrapper ?

Reply:
It doesn't seem to work. I think something to do with the fact that `value` here is local to this component, and so we'd have to use `props.initialValue` in the Wrapper, and it's not behaving correctly when I do. Maybe just leave as is for now?

Reply:
Sure

Reply:
Just realised this would show validation error that wont' necessarily be reflected in progress bar, which would require save, which might lead to some side effects. Need to have a deeper think about it.

The scenario I was thinking about, is this: Everything is valid, but user goes back a page, changes something that will affect validation in further page (page 3), loads page 3, invalid status would show on the question but is not saved (so not shown in progress bar etc..)